pinot-core/src/main/java/org/apache/pinot/core/data/readers/PinotSegmentRecordReader.java

```diff
@@ -81,25 +81,29 @@ public PinotSegmentRecordReader(File indexDir, @Nullable Schema schema, @Nullabl
         _schema = new SegmentMetadataImpl(indexDir).getSchema();
         Collection<String> columnNames = _schema.getColumnNames();
         _columnReaderMap = new HashMap<>(columnNames.size());
-        for (String columnName : columnNames) {
-          _columnReaderMap.put(columnName, new PinotSegmentColumnReader(_immutableSegment, columnName));
+        if (_numDocs > 0) {
```

Comment:
Instead of sprinkling `if (_numDocs > 0)`, can we branch out for `(if _numDocs == 0)` early on and handle empty segment separately?